### PR TITLE
feat: add sentry to track prod error

### PR DESCRIPTION
**Pull request by Innei**
### <samp>🤖 Generated by Copilot at 7308bba</samp>

This pull request adds Sentry integration for the Next.js application and the webpack builds, by installing and configuring various Sentry packages and plugins. It also updates the pnpm-lock.yaml file and the GitHub workflows to reflect the new dependencies and environment variables. It creates several configuration files, such as next.config.mjs, sentry.client.config.ts, sentry.edge.config.ts, sentry.properties, and sentry.server.config.ts, to customize and enable Sentry features and options for different parts of the application. The purpose of this pull request is to monitor and report errors and performance issues in the development and production environments.

### <samp>🤖 Generated by Copilot at 7308bba</samp>

> _To monitor errors and traces_
> _We added some Sentry packages_
> _And config files galore_
> _For client, edge, and server_
> _And also for webpack and Next.js_

### WHY
Error tracking

### <samp>🤖 Generated by Copilot at 7308bba</samp>

*  Enable Sentry integration for the development and production environments by adding four environment variables to the docker-build-push-dev and docker-build-push-prod workflows ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-456c1fec023a8b73522786b0b1f4408cb6495924b113f2d6a9867f169914b4d2L52-R57), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-b6bd6cf37a103202c2a51cfe576922ee8fb4adefcb3bcd3e11b83c4a9a875f62R52-R55))
*  Pass the Sentry configuration to the container by adding four arguments and environment variables to the Dockerfile ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R32-R39))
*  Provide a comprehensive and flexible configuration for the Next.js application and enable some features and integrations by adding a new file, `next.config.mjs`, that exports the Next.js configuration object and applies some plugins ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-18c049b08c4a0f5ab451c598aeb2c4848bb9d7877b51ca3e5effb94a225814d2R1-R147))
*  Install the packages that are required for the Sentry integration by adding two dependencies, `@sentry/nextjs` and `@sentry/webpack-plugin`, to the `pnpm-lock.yaml` file ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR357-R362))
*  Install the package that is required for the `@sentry/nextjs` plugin by adding a dependency, `@rollup/plugin-commonjs`, to the `pnpm-lock.yaml` file ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4202-R4219))
*  Install the package that is required for the `@rollup/plugin-commonjs` package by adding a dependency, `@rollup/pluginutils`, to the `pnpm-lock.yaml` file ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4257-R4271))
*  Install the packages that provide various functionalities and integrations for Sentry by adding several dependencies, such as `@sentry-internal/tracing`, `@sentry/browser`, `@sentry/cli`, and `@sentry/react`, to the `pnpm-lock.yaml` file ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4317-R4479))
*  Make some packages, such as `@types/estree`, `https-proxy-agent`, and `node-fetch`, regular dependencies instead of development dependencies by removing the dev flag from the `pnpm-lock.yaml` file ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5517), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8846), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8884), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11500))
*  Install the packages that are required by the Sentry dependencies by adding several dependencies, such as `chalk`, `cookie`, `estree-walker`, `glob`, `immediate`, `is-reference`, `lie`, `localforage`, `lru_map`, `magic-string`, `mkdirp`, `progress`, `rollup`, `stacktrace-parser`, and `type-fest`, to the `pnpm-lock.yaml` file ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR6410-R6417), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR6714-R6718), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8148-R8151), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8695-R8705), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR9173-R9176), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR9449-R9454), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR9987-R9992), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR10078-R10083), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR10201-R10204), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR10211-R10217), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR10816-R10822), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR11750-R11754), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR12552-R12559), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR12922-R12928), [link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR13523-R13527))
*  Enable Sentry monitoring and tracing for the client-side and the replay feature by adding a new file, `sentry.client.config.ts`, that exports the Sentry configuration for the client-side of the Next.js application ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-668b09e99b6037100b7526171c5b18292b5b22be0cf08b2a9dd55fbd64005dbaR1-R12))
*  Enable Sentry monitoring and tracing for the edge-side by adding a new file, `sentry.edge.config.ts`, that exports the Sentry configuration for the edge-side of the Next.js application ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-0579713d2867dd7b0b73196db50bf4197fd89375775158de6670415dff599032R1-R8))
*  Enable the `@sentry/cli` package to communicate with the Sentry API and to locate the Sentry CLI binary by adding a new file, `sentry.properties`, that contains the Sentry configuration for the command-line interface ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-48caab1bf3f29911795b9f38da0069e8958256328c771e0cd2a57c0f840ddd9dR1-R4))
*  Enable Sentry monitoring and tracing for the server-side by adding a new file, `sentry.server.config.ts`, that exports the Sentry configuration for the server-side of the Next.js application ([link](https://github.com/Crossbell-Box/xLog/pull/468/files?diff=unified&w=0#diff-1ab87e7d17ff99375cf43fd08d1fe605cff6b415d719ba8cb725ebd06f181cecR1-R8))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

**Comment by Innei** [comment]
Sentry integration is not tested, this code is copied from AFFiNE, thanks to @himself65.

[SENTRY_AUTH_TOKEN](https://github.com/toeverything/AFFiNE/blob/850cfe11877fea57534ae5e7bd847983278b63f8/apps/web/next.config.mjs#L201)

File: .env.example [hunk 1]
```diff
@@ -44,8 +44,18 @@ DB_SCHEMA=public
 
 # Prisma database connection
 
-DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DB_HOST}:${DB_PORT}/${POSTGRES_DB}?schema=${DB_SCHEMA}&sslmode=prefer&connection_limit=400&pool_timeout=30
+DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DB_HOST}:${DB_PORT}/${POSTGRES_DB}?schema=${DB_SCHEMA} &
```
Comment (DIYgod) [inline .env.example:47]: This string seems to be an unnecessary modification by prettier

**Review by DIYgod — COMMENTED** [review]
Thank you so much. I will wait for @sljeff to deploy a Sentry after his vacation so that I can test it.

File: .env.example (hunk 1 above)
Comment (Innei) [reply .env.example:47]: Oh, sorry.

**Comment by Innei** [comment]
Waiting for App Router finish.

**Comment by DIYgod** [comment]
Is there any progress on the support for App Router?

**Comment by Innei** [comment]
> 

There are no examples or demos in the official guide or repository. In App Router, error handle not like Page Router, it has no centralized processing pages or methods, but rather error pages scattered in the 'layout' dimension. I still don't know how to deal with it

**Comment by Innei** [comment]
Close this PR for now, and I will follow up on any progress later.

**Pull request closed without merging**